docs/migrate-to-nativecurlclient.md

```diff
@@ -0,0 +1,255 @@
+# Migrate from `Redmine\Client` to `Redmine\Client\NativeCurlClient`
+
+Since `php-redmine-api` v1.8.0 there is a new native cURL client
+`Redmine\Client\NativeCurlClient`. This guide will help you to migrate your code
+to this client.
+
+## 1. Use new client methods
+
+With the new interface `Redmine\Client\Client` there are now standarized methods
+for all clients. The new `Redmine\Client\NativeCurlClient` and the current `Redmine\Client` implementing this interface.
+
+### api() to getApi()
+
+Search in your code for the usage of `$client->api('issue')` and the magic
+getter like `$client->issue`. Then replace this calls with
+`$client->getApi('issue')`.
+
+```diff
+-$issue = $client->issue->show($issueId);
++$issue = $client->getApi('issue')->show($issueId);
+
+-$client->api('issue')->create($data);
++$client->getApi('issue')->create($data);
+```
+
+### getResponseCode() to getLastResponseStatusCode()
+
+Replace every call for `$client->getResponseCode()` with
+`$client->getLastResponseStatusCode()`.
+
+```diff
+-if ($client->getResponseCode() === 500)
++if ($client->getLastResponseStatusCode() === 500)
+{
+    throw new \Exception('Redmine call failed');
+}
+```
+
+### get() to requestGet()
+
+If you are using `$client->get()`, `$client->post()`, `$client->put()` or
+`$client->delete()` directly you will have to change your code. This methods
+parse a possible JSON or XML response but in future the parsing of the raw
+response body will be up to you.
+
+To help you with the parsing of the raw response the client interface introduces
+two new methods: `getLastResponseContentType()` and `getLastResponseBody()`.
+
+This example shows how you can parse the response body of a GET request.
+
+```diff
+-// We dont know if we will get json, xml or a string
+-$dataAsJsonOrXmlOrString = $this->client->get($path);
++$this->client->requestGet($path);
++// $body contains the raw http body of the response
++$body = $this->client->getLastResponseBody();
++
++// if response is XML, create a SimpleXMLElement object
++if ($body !== '' && 0 === strpos($this->client->getLastResponseContentType(), 'application/xml')) {
```

Comment:
ok for me, but maybe we should add another method that does exactly this, what do you think? It makes it more complex to use the lib that way...

Reply:
This code is copied from the `AbstractApi::get()` method. Parsing would only needed if someone don't use the `AbstractApi` but calls `$client->get()` directly. Personally I don't think that someone has done this but I think it's worth mentioning.